src/main/scala/org/allenai/ike/IkeBatchAPI.scala

```diff
@@ -0,0 +1,226 @@
+    val patternConfig = ConfigFactory.load(configFile)
+    val hugeLimit = 100000000
+    val extractionPatterns = patternConfig.getConfigList("ExtractionPatterns.patterns").asScala.map { config =>
+      config.getString("name") -> SearchRequest(Left(config.getString("pattern")), None, None, SearchConfig(hugeLimit, 1))
+    }.toMap
+    extractionPatterns
+  }
+
+  /** Peforms the search over the corpus given the specific pattern to look for (which was loaded from config)
+    *
+    *   // TODO: Should probably refactor to reuse parts of IKEToolWebApp, maybe some common pieces for both?
+    *   // TODO: lots of copypasta from IKEToolWebApp, gross
+    *
+    * @param patternName The pattern you are looking for
+    * @param req The SearchRequest object
+    * @param indexWithContent This is the name of hte index that holds hte content you care about. You need it
+    *                         because you want the filename and the sentence/content for this pattern's match
+    * @return A Seq of ApiExtractionResult case classes which hold the results
+    */
+  def doSearch(patternName: String, req: SearchRequest, indexWithContent: String): Seq[ApiExtractionResult] = {
+    import org.allenai.ike.JsonSerialization._
+
+    val query = SearchApp.parse(req).get
+
+    val (tables, patterns) = req.userEmail match {
+      case Some(userEmail) => (
+        Tablestore.tables(userEmail),
+        Tablestore.namedPatterns(userEmail)
+      )
+      case None => (Map.empty[String, Table], Map.empty[String, NamedPattern])
+    }
+
+    val interpolatedQuery = QueryLanguage.interpolateQuery(
+      query,
+      tables,
+      patterns,
+      word2vecPhrasesSearcher,
+      tableExpander
+    ).get
+
+    val foundResults = searchApps.map { case (name, searcher) => searcher.search(interpolatedQuery, req.config).get }
+    val qexpr = SearchApp.parse(req).get
+
+    // brakes the flatmap if empty
+    if (foundResults.head.isEmpty) {
+      return List[ApiExtractionResult]()
+    }
+
+    val extractions = foundResults.flatMap {
+      case rr =>
+        rr.map {
+          case blab =>
+            val docId = blab.sourceDocument.getOrElse(-1) // TODO: Should barf on -1
+
+            val extractedString = blab.matchWords.mkString(" ")
+
+            val extractionStart = blab.matchOffset.start
+            val extractionEnd = blab.matchOffset.end
+
+            // Lucene document is XML. Assuming one index where this documnent lives for now. TODO: sketchy...
+            val contentXML = XML.loadString(searchApps(indexWithContent).searcher.getContent(docId))
+            // get the words out, so we can make them into a string. We want the whole sentence, so need this
+            val sourceString = contentXML.text
+
+            // Again, assuming one index where this document lives for now... TODO: sketchy...
+            val sourceArticle = searchApps(indexWithContent).searcher.document(docId).get("fromInputFile").toString
```

Comment:
Ok, changed this so that the blab object selects the searcher (since it knows its own corpusName). So at least this one is cleaner.